### PR TITLE
Tests automated, secure connection & retry methods added

### PR DESCRIPTION
**Pull request by niveathika**
1. Tests automated - Normal as well as secure connections
2. Fix for secure connection - Including trust store and key store
3. Connect, Close and destroy method implementation(Siddhi-M12)

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java [hunk 1]
```diff
@@ -240,30 +286,24 @@ protected void init(TableDefinition tableDefinition, ConfigReader configReader)
...
-                        + "' is not successful due to " + e.getLocalizedMessage(), e);
-            }
-        } else {
-            MongoCursor<Document> existingIndicesIterator = this.getCollectionObject().listIndexes().iterator();
-            MongoTableUtils.checkExistingIndices(expectedIndexModels, existingIndicesIterator);
+        this.initialCollectionTest = false;
+        try {
+            this.mongoClient = new MongoClient(this.mongoClientURI);
+        } catch (MongoException e) {
+            throw new MongoTableException("Annotation 'Store' contains illegal value for " +
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java]: Why are we throwing this runtime exception to Siddhi? If you want initialization to fail shall we throw SiddhiAppCreationException? WDYT?

File: component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java [hunk 2]
```diff
@@ -496,4 +534,50 @@ protected CompiledCondition compileCondition(ConditionBuilder conditionBuilder)
         conditionBuilder.build(visitor);
         return new MongoCompiledCondition(visitor.getCompiledCondition(), visitor.getPlaceholders());
     }
+
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+        if (!this.initialCollectionTest) {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java:541]: format the code

**Review by tishan89 — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java (hunk 1 above)
Comment (niveathika) [reply component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java]: I've changed the initializing exceptions to SiddhiAppCreationException in d1c214c

File: component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java (hunk 2 above)
Comment (niveathika) [reply component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java:541]: Addressed the issues in d1c214c

**Pull request merged**